lib/model/bloc/startup/StartupViewModelProvider.dart

```diff
@@ -65,21 +68,34 @@ class StartupViewModelProvider implements ViewModelProvider<StartupViewModel> {
   }
 
   ChatPageViewModel _chatPageViewModel() {
-    return ChatPageViewModel(
-        _LocalisedAssets.onboardingFlow(), (data) {
-          final ChatResponseViewModel name = castOrNull<ChatResponseViewModel>(data["Name"]);
-          if (name != null){
-            _accountRepository.create("Test").then((account) {
-              final newProfile = ProfileEntity("test", name.value, MockImageEntity().build().urlProvider,plotInfo);
-              account.profileRepository.add(newProfile).then((profile){
-                account.profileRepository.switchTo(profile);
-                _refresh();
-              }) ;
-            });
-          }
-        },(data) {
-            //TODO: error case, should display a popup.
+    //TODO: Remove the Mock ID´s once implemented
+    final transformer = ChatResponseToPlotInfoTransformer();
+    return ChatPageViewModel(_LocalisedAssets.onboardingFlow(), (data) {
```

Comment:
This return does a lot of stuff inside, I think it should be split into some private methods

Reply:
split this up a bit, it will change in another PR too